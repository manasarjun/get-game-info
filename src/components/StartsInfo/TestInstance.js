import React from 'react';
import PropTypes from 'prop-types';

function TestInstance(props) {
  const { firstName, toggle, id, track } = props;
  return track.map(element => {
    return ((element.id === id && element.isOpen) ? <p>{firstName}</p> : null)
  });
  /*   return (
      <>
        {(toggle.id === id && toggle.isOpen) ? firstName : null}
      </>
    ) */
}

TestInstance.propTypes = {

}

export default TestInstance;

