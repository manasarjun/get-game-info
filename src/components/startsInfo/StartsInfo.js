import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

function StartsInfo({ starts }) {
  const [currentIndex, setCurrentIndex] = useState(null);
  const [isCollapsable, setIsCollapsable] = useState(false);


  const handleCollapsable = () => {
    setIsCollapsable(false);
  }

  const handleShow = (index) => () => {
    setCurrentIndex(index)
    setIsCollapsable(true);
  }

  return (
    <ul className='start-list'>{starts.map(
      (s, index) => (
        <li className='start-data' key={index}>
          <p>Start Number: {s.number}</p>
          <p>HorseName: {s.horse.name}</p>
          <p>Driver Name: {s.driver.firstName + '' + s.driver.lastName}</p>
          {index === currentIndex && isCollapsable ?
            <>
              <button onClick={handleCollapsable}><i className="material-icons">keyboard_arrow_up</i></button>
              <p>Trainer Name: {s.horse.trainer.firstName + '' + s.horse.trainer.lastName}</p>
              <p>Horse Father Name: {s.horse.pedigree.father.name}</p>
            </>
            :
            <button onClick={handleShow(index)}><i className="material-icons">keyboard_arrow_down</i></button>
          }
        </li>
      )
    )}
    </ul>
  )
}

StartsInfo.prototype = {
  starts: PropTypes.shape({
    number: PropTypes.number,
    horse: PropTypes.shape({
      name: PropTypes.string,
      pedigree: PropTypes.shape({
        father: PropTypes.shape({
          name: PropTypes.string,
        })
      })
    }),
    driver: PropTypes.shape({
      firstName: PropTypes.string,
      lastName: PropTypes.string,
    }),
    trainer: PropTypes.shape({
      firstName: PropTypes.string,
      lastName: PropTypes.string,
    }),


  })
}

export default StartsInfo;