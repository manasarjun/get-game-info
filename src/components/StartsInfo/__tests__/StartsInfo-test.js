import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import StartsInfo from '../StartsInfo';

describe.only('<StartsInfo />', () => {
  const baseProps = {
    number: 1,
    horse: {
      "name": "Deweykidumnhowe",
      "trainer": {
        "firstName": "Carina",
        "lastName": "Hezekielsson",
      },
      "pedigree": {
        "father": {
          "name": "Deweycheatumnhowe",
        },
      }
    },
    "driver": {
      "firstName": "Jens",
      "lastName": "Nilsson",
    }
  }

  it('renders correctly', () => {
    const wrapper = shallow(<StartsInfo />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
