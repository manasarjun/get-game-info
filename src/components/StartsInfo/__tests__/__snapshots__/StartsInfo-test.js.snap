// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartsInfo /> renders correctly 1`] = `
<ul
  className="start-list"
>
  <li
    className="start-data"
    key="0"
  >
    <p>
      Start Number: 
      1
    </p>
    <p>
      HorseName: 
      Deweykidumnhowe
    </p>
    <p>
      Driver Name: 
      JensNilsson
    </p>
    <button
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        keyboard_arrow_down
      </i>
    </button>
  </li>
</ul>
`;
